graphql_compiler/cost_estimation/cardinality_estimator.py

```diff
@@ -82,57 +82,126 @@ def _get_parent_and_child_name_from_edge(schema_graph, child_location):
     return parent_name_from_edge, child_name_from_edge
 
 
-def _estimate_children_per_parent(
-    schema_graph, statistics, query_metadata, parameters, child_location, parent_location
+def _get_outbound_and_inbound_names_from_locations(query_metadata, child_location, parent_location):
+    """Get the out and in classnames of a location and its parent from the last edge information."""
+    edge_direction, _ = _get_last_edge_direction_and_name_to_location(child_location)
+    parent_name_from_location = query_metadata.get_location_info(parent_location).type.name
+    child_name_from_location = query_metadata.get_location_info(child_location).type.name
+    if edge_direction == INBOUND_EDGE_DIRECTION:
+        outbound_vertex_name = child_name_from_location
+        inbound_vertex_name = parent_name_from_location
+    elif edge_direction == OUTBOUND_EDGE_DIRECTION:
+        outbound_vertex_name = parent_name_from_location
+        inbound_vertex_name = child_name_from_location
+    return outbound_vertex_name, inbound_vertex_name
```

Comment:
nit: add an else-block here to check that edge_direction is one of inbound or outbound.

same for `_get_parent_and_child_name_from_edge` (the method above)